cluster-agent/controllers/argoproj.io/application_controller_test.go

```diff
@@ -0,0 +1,556 @@
+package argoprojio_test
+
+import (
+	"context"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/managed-gitops/backend/eventloop/eventlooptypes"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	appv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/redhat-appstudio/managed-gitops/backend-shared/config/db"
+	sharedutil "github.com/redhat-appstudio/managed-gitops/backend-shared/util"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend/apis/managed-gitops/v1alpha1"
+	argo "github.com/redhat-appstudio/managed-gitops/cluster-agent/controllers/argoproj.io"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	name      = "my-application"
+	namespace = "argocd"
+	dbID      = "databaseID"
+)
+
+var _ = Describe("Application Controller", func() {
+	Context("Application Controller Test", func() {
+		var ctx context.Context
+		var k8sClient *sharedutil.ProxyClient
+		var managedEnvironment *db.ManagedEnvironment
+		var gitopsEngineInstance *db.GitopsEngineInstance
+		var dbQueries db.AllDatabaseQueries
+		var guestbookApp *appv1.Application
+		var reconciler argo.ApplicationReconciler
+		var err error
+
+		BeforeEach(func() {
+			scheme, argocdNamespace, kubesystemNamespace, workspace, err := eventlooptypes.GenericTestSetup()
+			Expect(err).To(BeNil())
+
+			err = appv1.AddToScheme(scheme)
+			Expect(err).To(BeNil())
+
+			gitopsDepl := &managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-gitops-depl",
+					Namespace: workspace.Name,
+					UID:       uuid.NewUUID(),
+				},
+			}
+
+			// Fake kube client.
+			k8sClientOuter := fake.NewClientBuilder().WithScheme(scheme).WithObjects(gitopsDepl, workspace, argocdNamespace, kubesystemNamespace).Build()
+			k8sClient = &sharedutil.ProxyClient{
```

Comment:
Looks like you can remove the `sharedutil.ProxyClient` here, because the test code doesn't use it's functionality (AFAICT).